ui/src/components/dataset/DatasetNameInput.vue

```diff
@@ -0,0 +1,45 @@
+<template>
```

Comment:
Can be used when creating a new dataset from the portal.

Allows:
- typing in a dataset name in an empty input field
- loading the input field with a pre-populated dataset name, which can then be edited
- updated input text is propagated to the parent component